pkg/web_app/lib/src/mobile_nav.dart

```diff
@@ -33,21 +34,62 @@ void _setEventForDetailMetadataToggle() {
   // Stored x,y coordinate of the scroll position at the time of the opening of metadata.
   int? origX, origY;
 
+  var isVisible = false;
+  // ignore: cancel_subscriptions
+  StreamSubscription? stateSubscription;
+  final currentTitle = document.head?.querySelector('title')?.text?.trim();
+  final currentUrl = window.location.toString();
   document.querySelectorAll('.detail-metadata-toggle').forEach((e) {
     e.onClick.listen((_) async {
-      document.querySelector('.detail-wrapper')?.classes.toggle('-active');
-      document.querySelector('.detail-metadata')?.classes.toggle('-active');
-      await window.animationFrame;
-      if (origX == null) {
-        // store scroll position and scroll to the top
-        origX = window.scrollX;
-        origY = window.scrollY;
-        window.scrollTo(0, 0);
+      Future<void> toggle() async {
+        isVisible = !isVisible;
+
+        document.querySelector('.detail-wrapper')?.classes.toggle('-active');
+        document.querySelector('.detail-metadata')?.classes.toggle('-active');
+        await window.animationFrame;
+        if (origX == null) {
+          // store scroll position and scroll to the top
+          origX = window.scrollX;
+          origY = window.scrollY;
+          window.scrollTo(0, 0);
+        } else {
+          // restore scroll position
+          window.scrollTo(origX, origY);
+          origX = null;
+        }
+      }
+
+      // when activating, set the current state to hide the view, and the next
+      // state to activate  the view
+      if (!isVisible) {
+        await toggle();
+        window.history.replaceState({
+          'type': 'detail-metadata',
+          'url': currentUrl,
+          'visible': false,
+        }, '', null);
+        window.history.pushState({
+          'type': 'detail-metadata',
+          'url': currentUrl,
+          'visible': true,
+        }, currentTitle ?? '', null);
       } else {
-        // restore scroll position
-        window.scrollTo(origX, origY);
-        origX = null;
+        window.history.back();
       }
+
+      // only listen to state events after the first initialization
+      stateSubscription ??= window.onPopState.listen((event) async {
```

Comment:
This won't work well, if you:
 1. Navigate to `pub.dev/packages/retry`
 2. Toggle the "details / meta-data" view
 3. Click the repository link (navigating to `github.com/...`)
 4. Click the back button.
 5. The "details / meta-data" view will not be active, leaving users disappointed.

Am I wrong? (I haven't tested the theory)

We could say that (5) is an acceptable downside.

------------------

We could also use `pub.dev/packages/retry#details` to toggle the "details / meta-data" view. Storing the state in `document.location.hash`.

We might be able to use the [`hashchange` event](https://developer.mozilla.org/en-US/docs/Web/API/Window/hashchange_event) then. Though it might be just as complicated.

------------------

An alternative is to create a separate page: `pub.dev/packages/<package>/details` that contains the sidebar. It'll be a bit sad, because it won't work when you resize the window.

Reply:
I've deployed the PR to staging, and it seems to me that clicking through 1-5 is working as expected, at least on Chrome.